src/__tests__/preset-by-number.test.ts

```diff
@@ -0,0 +1,115 @@
+import { describe, expect, test } from '@jest/globals'
+import { parsePresetVariableOption } from '../preset-by-number.js'
+import { CompanionOptionValues } from '@companion-module/base'
+import { PresetRecall } from '../camera/commands.js'
+import { prettyBytes } from '../visca/utils.js'
+import { MockContext } from './mock-context.js'
+import type { MatcherFunction } from 'expect'
+import { Command } from '../visca/command.js'
+
+function optionsWithPresetOption(text: string): CompanionOptionValues {
+	return { val: text }
+}
+
+/**
+ * Custom jest matcher to check that options are not a string and that the command bytes match
+ * @param actual
+ * @param command
+ * @param expected_bytes
+ * @returns
+ */
+const toBeExpectedCommandBytes: MatcherFunction<[command: Command, expected_bytes: string]> = function (
+	actual,
+	command,
+	expected_bytes
+) {
+	if (typeof actual === 'string') {
+		throw new TypeError('Input showuld be of type CompanionOptionValues not string')
+	}
+
+	const commandBytes = command.toBytes(actual as CompanionOptionValues)
+	const pass = prettyBytes(commandBytes) === expected_bytes
```

Comment:
Can't you just add `: CompanionOptionValues` next to `actual` in the function argument list?  We pretty much should never see `as` in actual code.

Comparing `prettyBytes` output is kinda skeevy compared to making the expected byte list an array of numbers.  But also it's test code, and sometimes I need to remember that it is okay for test code to be messier, as long as it's not unreadably messy.

Reply:
Unfortunately no, it's not compatible with the matcher definition.  Otherwise would be happy to

Reply:
I pulled the PR code and experimented with it -- `actual` is `unknown` here (and so you have to type-test it to do anything with it).  It's easy to test for a string.  But it's not actually trivial to use `actual` as a `CompanionOptionValues`.  I'm not even sure it's possible to do it without casting -- casting that is entirely avoidable if we just test the values directly, rather than defining a matcher.

At this point I think we're better off just eschewing a custom matcher: write normal, typed functions to do the tests, along lines something like:

```ts
function expectResolvesToCommandBytes(
    result: string | CompanionOptionValues,
    command: Command,
    expectedBytes: readonly number[],
): void {
    if (typeof result === 'string') {
        throw new TypeError('Result should not have been an error, instead got: ' + result)
    }

    const commandBytes = command.toBytes(result)
    expect(commandBytes.length).toBe(expectedBytes.length)
    for (let i = 0; i < commandBytes.length; i++) {
        expect(commandBytes[i]).toBe(expectedBytes[i])
    }
}

function expectIsErrorString(
    result: string | CompanionOptionValues,
): void {
    if (typeof result !== 'string') {
        throw new TypeError('Result should have been an error')
    }
}
```

And then have callers specify the command bytes as numbers, like `[0x81, 0x01, 0x04, 0x3F, 0x02, 0x25, 0xFF]`.